### PR TITLE
ptunnel-ng: build script

### PR DESCRIPTION
**Pull request by oofnikj**
Adding build.sh for `ptunnel-ng`, a tool to tunnel TCP traffic over ICMP. https://github.com/lnslbrty/ptunnel-ng/

Results in `Bad system call` when running in un-rooted mode. Workaround using `proot -0`. I'm not an expert in C but I'm thinking perhaps this can be patched to not require `proot`?

Alternatively this can be moved to `termux-root-packages`, but I would prefer it to be made available to un-rooted devices since it is still useful for them.

### CODE REVIEW

**Review by xeffyr — CHANGES_REQUESTED** [review]
> Results in Bad system call when running in un-rooted mode. Workaround using proot -0.

Privilege dropping has to be removed as setuid/setgid and related system calls are not allowed on Android OS due to seccomp filter.

https://github.com/lnslbrty/ptunnel-ng/blob/af15e7f597ad03d541a43199b29e422971e48ed8/src/ptunnel.c#L595-L597

**Comment by oofnikj** [comment]
> > Results in Bad system call when running in un-rooted mode. Workaround using proot -0.
> 
> Privilege dropping has to be removed as setuid/setgid and related system calls are not allowed on Android OS due to seccomp filter.
> 
> https://github.com/lnslbrty/ptunnel-ng/blob/af15e7f597ad03d541a43199b29e422971e48ed8/src/ptunnel.c#L595-L597

Thanks, added a patch to remove these calls. Works now without `proot`.

**Comment by landfillbaby** [comment]
i think that patch should probably remove the preceding `if` too or the entire program flow might get messed up

File: packages/ptunnel-ng/0001-remove-calls-to-setuid-and-setgid.patch [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ src/ptunnel.c | 5 -----
+ 1 file changed, 5 deletions(-)
+
+diff --git a/src/ptunnel.c b/src/ptunnel.c
+index d70edce..6d64519 100644
+--- a/src/ptunnel.c
++++ b/src/ptunnel.c
+@@ -553,11 +553,6 @@ void* pt_proxy(void *args) {
+ #else
+ 	if (opts.uid || opts.gid)
```
Comment (landfillbaby) [inline packages/ptunnel-ng/0001-remove-calls-to-setuid-and-setgid.patch:16]: should probably remove this line too, even if the #if condition is always false it just doesn't look right

Comment (landfillbaby) [reply packages/ptunnel-ng/0001-remove-calls-to-setuid-and-setgid.patch:16]: https://github.com/lnslbrty/ptunnel-ng/blob/v1.42/src/ptunnel.c#L551-L556
i just checked and yes you definitely need to remove the preceding #if
note the lack of a closing semicolon in either condition

Comment (landfillbaby) [reply packages/ptunnel-ng/0001-remove-calls-to-setuid-and-setgid.patch:16]: tl;dr the patch should remove lines 551 to 560, it currently only removes lines 556 to 560

Comment (landfillbaby) [reply packages/ptunnel-ng/0001-remove-calls-to-setuid-and-setgid.patch:16]: also if it were up to me i would remove all references to opts.uid and opts.gid, including the -u command-line option

**Comment by landfillbaby** [comment]
brb making a better patch using the preexisting `#if`s

**Comment by landfillbaby** [comment]
[nosetugid.patch.txt](https://github.com/termux/termux-packages/files/5340730/nosetugid.patch.txt)

**Comment by oofnikj** [comment]
Thanks @landfillbaby, just tested your patch. Surprised mine even compiled, let alone worked at all!
Updated the PR with the new patch.

**Pull request merged**